### PR TITLE
refactor(frontend): use typescript hell :burn:

### PR DESCRIPTION
**Pull request by lionelB**

### CODE REVIEW

File: targets/frontend/next-env.d.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+/// <reference types="next" />
+/// <reference types="next/types/global" />
```
Comment (m-maillot) [inline targets/frontend/next-env.d.ts:2]: It's marked as mandatory in the next.js doc. 
https://nextjs.org/docs/basic-features/typescript

Why did you comment on these lines?

Comment (lionelB) [reply targets/frontend/next-env.d.ts:2]: I thought it was created by next. I will look to the format

File: targets/frontend/src/components/button/index.js.txt [hunk 2]
```diff
@@ -125,14 +125,15 @@ const OutlineButton = React.forwardRef(function _OutlineButton(
 });
 OutlineButton.propTypes = buttonPropTypes;
 
-export const IconButton = React.forwardRef(function _IconButton(
-  { variant = "primary", size = "large", sx, ...props },
-  ref
-) {
+export const IconButton = function _IconButton({
```
[the remark this answers is not in the document]
Comment (lionelB) [reply targets/frontend/src/components/button/index.js.txt:128]: true :)

File: targets/frontend/next-env.d.ts (hunk 1 above)
Comment (m-maillot) [reply targets/frontend/next-env.d.ts:2]: We have the same on cdtn project... indeed it's auto generated :)

**Pull request merged**